examples/plot_sleep_staging_sequences.py

```diff
@@ -0,0 +1,382 @@
+# the classifier). We do this by creating a collection of SequenceDatasets:
+
+import numpy as np
+from braindecode.datasets import get_sequence_dataset
+
+seq_len = 5  # Sequences of 5 consecutive windows
+step_len = 1  # Maximally overlapping sequences
+# Use label of center window in the sequence
+label_transform = lambda x: x[np.ceil(len(x) / 2).astype(int)]  # noqa: E731
+windows_dataset = get_sequence_dataset(
+    windows_dataset, seq_len=seq_len, step_len=step_len,
+    label_transform=label_transform)
+
+
+######################################################################
+# Split dataset into train and valid
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+#
+
+######################################################################
+# We can easily split the dataset using additional info stored in the
+# `description` attribute of :class:`braindecode.datasets.BaseDataset`,
+# in this case using the ``subject`` column. Here, we split the examples per subject.
+#
+
+splitted = windows_dataset.split('subject')
+train_set = splitted['0']
+valid_set = splitted['1']
+
+# Print number of examples per class
+print(train_set.datasets[0].windows)
+print(valid_set.datasets[0].windows)
+
+
+######################################################################
+# Create model
+# ------------
+#
+
+######################################################################
+# We can now create the deep learning model. In this tutorial, we use the sleep
+# staging architecture introduced in [1]_, which is a four-layer convolutional
+# neural network.
+#
+
+import torch
+from torch import nn
+from braindecode.util import set_random_seeds
+from braindecode.models import SleepStagerChambon2018
+
+cuda = torch.cuda.is_available()  # check if GPU is available
+device = 'cuda' if torch.cuda.is_available() else 'cpu'
+if cuda:
+    torch.backends.cudnn.benchmark = True
+# Set random seed to be able to reproduce results
+set_random_seeds(seed=87, cuda=cuda)
+
+n_classes = 5
+# Extract number of channels and time steps from dataset
+n_channels = train_set[0][0].shape[1]
+input_size_samples = train_set[0][0].shape[2]
+
+
+class TimeDistributedNet(nn.Module):
+    """Model features are extracted for multiple windows then concatenated.
+    """
+    def __init__(self, feat_extractor, seq_len, n_classes, dropout=0.25):
+        super().__init__()
+        self.feat_extractor = feat_extractor
+        self.clf = nn.Sequential(
+            nn.Dropout(dropout),
+            nn.Linear(feat_extractor.len_last_layer * seq_len, n_classes)
+        )
+
+    def forward(self, x):
+        """
+        Parameters
+        ----------
+        x : torch.Tensor
+            Input sequence of windows, of shape (batch_size, seq_len,
+            n_channels, n_times).
+        """
+        feats = [self.feat_extractor.embed(x[:, i]) for i in range(x.shape[1])]
+        feats = torch.stack(feats, dim=0).transpose(0, 1).flatten(start_dim=1)
```

Comment:
So the batch dimension is not returned as the first dimension? Why do we need this transpose before flatten?

Reply:
I just needed to change to `dim=1` and I could remove the transpose.